utils/obsproc/rtofs/RTOFSOb.cc

```diff
@@ -0,0 +1,160 @@
+#include "RTOFSOb.h"
+
+#include <stdio.h>
+
+#include <string>
+using std::string;
+
+#include <iostream>
+using std::endl;
+using std::cerr;
+
+#include <fstream>
+using std::ofstream;
+
+#include "util.h"
+
+
+namespace rtofs
+{
+
+
+RTOFSOb::
+    RTOFSOb(int n, int mx_lvl, int vrsn):
+        n(n),
+        mx_lvl(mx_lvl),
+        vrsn(vrsn)
+{
+    dtg = new std::time_t[n];
+    allocate();
+}
+
+
+
+void
+RTOFSOb::
+    allocate()
+{
+    lat = new float[n];
+    lon = new float[n];
+    btm = new float[n];
+    ls = new int[n];
+    lt = new int[n];
+    sal_typ = new int[n];
+    sqc = new float[n];
+    tmp_typ = new int[n];
+    tqc = new float[n];
+
+    lvl = new float * [n];
+    sal = new float * [n];
+    sal_err = new float * [n];
+    sprb = new float * [n];
+    tmp = new float * [n];
+    tmp_err = new float * [n];
+    clm_sal = new float * [n];    // skipped?
+    tprb = new float * [n];
+    cssd = new float * [n];
+    clm_tmp = new float * [n];    // skipped?
+    ctsd = new float * [n];
+    flg = new int * [n];
```

Comment:
Is there a reason not to give these varaibles more transparent names?

Reply:
These are coming from the f77 side @AndrewEichmann-NOAA , it probably makes sense to keep it that way but could we add comments detailing what they are @givelberg ?

Reply:
We could, but we need to ask Zulema & Co.
The variables that have a comment "skipped?" are arrays that exist in the binary file, but were not read in the fortran code.